gio-subclass/src/auto/action.rs

```diff
@@ -0,0 +1,274 @@
+
+}
+
+any_impl!(ActionImpl);
+
+pub trait ActionImplStatic<T: ObjectType>: 'static {
+    fn get_impl<'a>(&self, imp: &'a T::ImplType) -> &'a ActionImpl;
+}
+
+struct ActionStatic<T: ObjectType>{
+    imp_static: *const ActionImplStatic<T>
+}
+
+// FIXME: Boilerplate
+
+unsafe extern "C" fn action_activate<T: ObjectType>
+(gptr: *mut gio_ffi::GAction, parameter: *mut glib_ffi::GVariant)
+{
+    floating_reference_guard!(gptr);
+    let klass = &**(gptr as *const *const ClassStruct<T>);
+    let interface_static = klass.get_interface_static(gio_ffi::g_action_get_type())
+                                     as *const ActionStatic<T>;
+    let instance = &*(gptr as *const T::InstanceStructType);
+    let imp = instance.get_impl();
+    let imp = (*(*interface_static).imp_static).get_impl(imp);
+    let wrap = from_glib_borrow(gptr);
+    imp.activate(&wrap, (if parameter.is_null() { None } else { Some(from_glib_none(parameter)) }).as_ref())
+}
+
+unsafe extern "C" fn action_change_state<T: ObjectType>
+(gptr: *mut gio_ffi::GAction, value: *mut glib_ffi::GVariant)
+{
+    floating_reference_guard!(gptr);
+    let klass = &**(gptr as *const *const ClassStruct<T>);
+    let interface_static = klass.get_interface_static(gio_ffi::g_action_get_type())
+                                     as *const ActionStatic<T>;
+    let instance = &*(gptr as *const T::InstanceStructType);
+    let imp = instance.get_impl();
+    let imp = (*(*interface_static).imp_static).get_impl(imp);
+    let wrap = from_glib_borrow(gptr);
+    imp.change_state(&wrap, &from_glib_none(value))
+}
+
+unsafe extern "C" fn action_get_enabled<T: ObjectType>
+(gptr: *mut gio_ffi::GAction) -> gboolean
+{
+    floating_reference_guard!(gptr);
+    let klass = &**(gptr as *const *const ClassStruct<T>);
+    let interface_static = klass.get_interface_static(gio_ffi::g_action_get_type())
+                                     as *const ActionStatic<T>;
+    let instance = &*(gptr as *const T::InstanceStructType);
+    let imp = instance.get_impl();
+    let imp = (*(*interface_static).imp_static).get_impl(imp);
+    let wrap = from_glib_borrow(gptr);
+    imp.get_enabled(&wrap).to_glib()
+}
+
+unsafe extern "C" fn action_get_name<T: ObjectType>
+(gptr: *mut gio_ffi::GAction) -> *const c_char
+{
+    floating_reference_guard!(gptr);
+    let klass = &**(gptr as *const *const ClassStruct<T>);
+    let interface_static = klass.get_interface_static(gio_ffi::g_action_get_type())
+                                     as *const ActionStatic<T>;
+    let instance = &*(gptr as *const T::InstanceStructType);
+    let imp = instance.get_impl();
+    let imp = (*(*interface_static).imp_static).get_impl(imp);
+    let wrap = from_glib_borrow(gptr);
+
+    unsafe extern "C" fn destroy(p: glib_ffi::gpointer){
+        glib_ffi::g_free(p);
+    };
+
+    match imp.get_name(&wrap)
+    {
+        Some(t) => {
+            let ret = t/*Not checked*/.to_glib_none().0;
```

Comment:
This needs to be `to_glib_full()` so that the return value stays valid forever until you free it